src/utils/localization-helper.js

```diff
@@ -0,0 +1,49 @@
+import { AsyncStorage } from 'react-native';
+import moment from 'moment/min/moment-with-locales.min';
+
+import { common } from 'config';
+import I18n from 'locale';
+
+export const translate = (key, locale, interpolation = null) =>
+  I18n.t(key, { locale, ...interpolation });
+
+export const getLocale = () => {
+  const locale = I18n.locale.toLowerCase();
+  const specialLocales = {
+    'zh-hans': 'zh-cn',
+    'zh-hans-cn': 'zh-cn',
+    'zh-hans-sg': 'zh-cn',
+    'zh-hans-hk': 'zh-hk',
+    'zh-hant': 'zh-tw',
+    'zh-hant-tw': 'zh-tw',
+    'zh-hant-hk': 'zh-tw',
+  };
+
+  return specialLocales[locale] || locale;
```

Comment:
@lex111 We still need something like `I18n.locale.substr(0, 2)`. Maybe separate by '-' and remove the last part.

If I change my iPhone device language to French, `I18n.locale` returns `fr-zh`. But it is not supported by `moment`.

Reply:
@chinesedfan No, it is not necessary, it was a mistake. For example, in the case of `pt-br`, it will be just `pt`, but we need `pt-br`, because translation is only for him.

> If I change my iPhone device language to French, I18n.locale returns fr-zh. But it is not supported by moment.

There is a [check](https://github.com/gitpoint/git-point/pull/479/files#diff-aa84d914cd1a32be6d7b8abcb41cf186R37) if there is such a locale in `moment`, and if not, then the default locale (English).

Reply:
@lex111 I mean, `fr-zh` should also result in `fr`, instead of `en`.

Reply:
@chinesedfan I fixed it, look now.